apps/experiments/decorators.py

```diff
@@ -86,19 +87,24 @@ def my_view(request, team_slug, experiment_id, session_id):
 
     @wraps(view)
     def _inner(request, *args, **kwargs):
+        allow_superuser_access = False
         if request.user.is_authenticated:
             if request.experiment_session.participant.user_id == request.user.id:
                 return view(request, *args, **kwargs)
-            elif request.team_membership and request.user.has_perm("chat.view_chat"):
-                return view(request, *args, **kwargs)
+            elif request.team_membership:
+                if request.user.has_perm("chat.view_chat"):
+                    return view(request, *args, **kwargs)
+            else:
+                # If the user is authed but not a team member, initiate superuser access checks
+                allow_superuser_access = True
 
         try:
             access_value = get_chat_session_access_cookie_data(request)
         except (signing.BadSignature, KeyError):
-            raise Http404()
+            raise TeamAccessDenied() if allow_superuser_access else Http404()
```

Comment:
Does `request.user.is_superuser` not work?

Reply:
I suppose it would

Reply:
e217c8511db1db02dfd0c25f0d01fc9d7645a6fc